ci/install-musl.sh

```diff
@@ -64,13 +64,62 @@ esac
 cd ..
 rm -rf "$musl"
 
-# Download, configure, build, and install musl-sanitized kernel headers:
-kernel_header_ver="4.19.88"
-curl --retry 5 -L \
-    "https://github.com/sabotage-linux/kernel-headers/archive/v${kernel_header_ver}.tar.gz" |
-    tar xzf -
+# Download, configure, build, and install musl-sanitized kernel headers.
+
+# Alpine follows stable kernel releases, 3.20 uses Linux 6.6 headers.
+alpine_version=3.20
+alpine_git=https://gitlab.alpinelinux.org/alpine/aports
+
+# This routine piggybacks on: https://git.alpinelinux.org/aports/tree/main/linux-headers?h=3.20-stable
+git clone -n --depth=1 --filter=tree:0 -b "${alpine_version}-stable" "$alpine_git"
 (
-    cd "kernel-headers-${kernel_header_ver}"
-    make ARCH="${kernel_arch}" prefix="/musl-${musl_arch}" install -j4
+   cd aports
+   git sparse-checkout set --no-cone main/linux-headers
+   git checkout
+
+   cd main/linux-headers
+   cp APKBUILD APKBUILD.vars
+   {
+   echo "printf \"\$source\" > alpine-source"
+   # shellcheck disable=SC2028
+   echo "printf \"\$_kernver\n\" > alpine-kernver"
+   # shellcheck disable=SC2028
+   echo "printf \"\$pkgver\n\" > alpine-pkgver"
+   # shellcheck disable=SC2028
+   echo "printf \"\$sha512sums\n\" > alpine-sha512sums"
+   } >> APKBUILD.vars
+   sh APKBUILD.vars
+
+   kernel_version=$(tr -d "[:space:]" < alpine-kernver)
+   pkg_version=$(tr -d "[:space:]" < alpine-pkgver)
+
+   urls=$(grep -o 'https.*' alpine-source)
+   kernel=""
+   patch=""
+   for url in $urls; do
+      base=$(basename "$url")
+      curl --retry 5 -L "$url" > "$base"
+      case $base in
+         linux-*) kernel=$base;;
+         patch-*) patch=$base;;
+      esac
+   done
```

Comment:
Does the following work instead?

```suggestion
   urls=$(grep -o 'https.*' alpine-source)
   for url in $urls; do
      base=$(basename "$url")
      
      # We only need to download `linux` and the patch archive, not individual
      # patches.
      case "$base" in
         linux-*) kernel="$base";;
         patch-*) patch="$base";;
         *) continue;;
      esac

      curl --retry 5 -L "$url" -o "$base"
      grep "$base" alpine-sha512sums >> sha-check
   done
```

Then remove the `grep "$kernel/patch" alpine-sha512sums >> sha-check` lines below. This makes sure we don't download/use anything that we don't verify the sha of.

If we do actually need to download all the files (`revert-broken-uapi.patch` and `0003-remove-inclusion-of-sysinfo.h-in-kernel.h.patch`), then remove `*) continue` and `>> sha-check`, instead just `sha512sum -c alpine-sha512sums`.

Reply:
We need the latter. I added an explicit check to avoid downloading files that do not contain a checksum.